api/coreservice.go

```diff
@@ -1430,19 +1430,19 @@ func (core *coreService) EstimateExecutionGasConsumption(ctx context.Context, sc
 	sc.SetGasPrice(big.NewInt(0))
 	blockGasLimit := core.bc.Genesis().BlockGasLimit
 	sc.SetGasLimit(blockGasLimit)
-	enough, receipt, err := core.isGasLimitEnough(ctx, callerAddr, sc)
+	enough, ret, receipt, err := core.isGasLimitEnough(ctx, callerAddr, sc)
 	if err != nil {
 		return 0, status.Error(codes.Internal, err.Error())
 	}
 	if !enough {
-		if receipt.ExecutionRevertMsg() != "" {
-			return 0, status.Errorf(codes.Internal, fmt.Sprintf("execution simulation is reverted due to the reason: %s", receipt.ExecutionRevertMsg()))
+		if receipt != nil && len(receipt.ExecutionRevertMsg()) > 0 {
+			return 0, newRevertError(ret)
 		}
 		return 0, status.Error(codes.Internal, fmt.Sprintf("execution simulation failed: status = %d", receipt.Status))
```

Comment:
the return type of error should be consistent

Reply:
`newRevertError`  In addition to being compatible with the error interface, it also provides the interface data required by JSON-RPC. code from go-ethereum